sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -47,7 +47,7 @@ import org.apache.spark.sql.types._
 object TypeCoercion {
 
   def typeCoercionRules(conf: SQLConf): List[Rule[LogicalPlan]] =
-    InConversion(conf) ::
+    PredicateSubqueryConversion(conf) ::
```

Comment:
@francis0407 So this function also handles the Inset coercion along with the predicate subquery , right ? I think we should name it correctly or else handle the inset coercion in a different function ?

@maropu any preference here ?

Reply:
Yea, we should change to better name.

Reply:
I'm not sure what to use, since the name is so long here. `PredicateSubqueryAndInConversion` may be too long to use.